doc/dev/mds_internals/quiesce.rst

```diff
@@ -0,0 +1,86 @@
+MDS Quiesce Protocol
+====================
+
+The MDS quiesce protocol is a a mechanism for "quiescing" (quieting) a
+subvolume in the file system, stopping all write (and most read) I/O.  A
+subvolume is a subtree in the file system marked with the "ceph.dir.subvolume"
+flag (like the mgr volumes plugin subvolumes).
+
+The purpose of this API is to prevent multiple clients from interleaving reads
+and writes across an eventually consistent snapshot barrier where out-of-band
+communication exists between clients. This communication can lead to clients
+wrongly believing they've reached a checkpoint that is mutually recoverable to
+via a snapshot.
+
+Mechanism
+---------
+
+The MDS quiesces I/O using a new ``quiesce_subvolume`` internal request that
+obtains appropriate locks on the root of a subvolume and then launches a series
+of sub-requests for locking other inodes in the subvolume. The locks obtained
+will force clients to release caps and in-progress client/mds requests to
+complete.
+
+The sub-requests launched are ``quiesce_subvolume_inode`` internal requests
+which simply lock the inode, if auth. Generally, these are read locks on each
+Inode metadata lock but the ``filelock`` is xlocked (exclusively locked)
```

Comment:
Other instances of "inode" here are not capitalized.  Suggest consistency one way or the other.